chain/xrp/client/client.go

```diff
@@ -0,0 +1,992 @@
+	serializedTxInputHex := hex.EncodeToString(serializedTxInputBytes)
+	serializedTxInputHexBytes := []byte(serializedTxInputHex)
+
+	submitRequest := &SubmitRequest{
+		Method: "submit",
+		Params: []SubmitParamEntry{
+			{
+				TxBlob: string(serializedTxInputHexBytes),
+			},
+		},
+	}
+
+	var submitResponse SubmitResponse
+	err = client.Send(MethodPost, submitRequest, &submitResponse)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// FetchTxInfo Returns transaction info - new endpoint
+func (client *Client) FetchTxInfo(ctx context.Context, txHash xc.TxHash) (xclient.TxInfo, error) {
+	txInfo, err := client.GetTxInfo(ctx, txHash)
+	if err != nil {
+		return xclient.TxInfo{}, err
+	}
+
+	return txInfo, nil
+}
+
+func (client *Client) FetchLegacyTxInfo(ctx context.Context, txHash xc.TxHash) (xc.LegacyTxInfo, error) {
+	panic("implement me")
+}
+
+func (client *Client) GetTxInfo(ctx context.Context, txHash xc.TxHash) (xclient.TxInfo, error) {
+	txRequest := &TransactionRequest{
+		Method: "tx",
+		Params: []TransactionParamEntry{
+			{
+				Transaction: txHash,
+				Binary:      false,
+			},
+		},
+	}
+
+	var txResponse TransactionResponse
+	err := client.Send(MethodPost, txRequest, &txResponse)
+	if err != nil {
+		return xclient.TxInfo{}, err
+	}
+
+	ledgerRequest := LedgerRequest{
+		Method: "ledger",
+		Params: []LedgerParamEntry{
+			{
+				LedgerIndex: "current",
+			},
+		},
+	}
+
+	var ledgerResponse LedgerResponse
+	err = client.Send(MethodPost, ledgerRequest, &ledgerResponse)
+	if err != nil {
+		return xclient.TxInfo{}, err
+	}
+
+	name := xclient.TransactionName(string(client.Asset.GetChain().Chain) + txResponse.Result.Hash)
+
+	block := xclient.NewBlock(uint64(txResponse.Result.LedgerIndex), txResponse.Result.Hash, time.Unix(txResponse.Result.Date, 0))
+
+	confirmations := ledgerResponse.Result.LedgerCurrentIndex - txResponse.Result.Sequence
+
+	var errMsg *string
+	if txResponse.Result.Status == "error" {
+		msg := "transaction failed"
+		errMsg = &msg
+	}
+
+	txInfo := xclient.TxInfo{
+		Name:          name,
+		Hash:          txResponse.Result.Hash,
+		Chain:         client.Asset.GetChain().Chain,
+		Block:         block,
+		Transfers:     []*xclient.Transfer{},
+		Fees:          []*xclient.Balance{},
+		Confirmations: uint64(confirmations),
+		Error:         errMsg,
+	}
+
+	affectedNodes := txResponse.Result.Meta.AffectedNodes
+	tf := xclient.NewTransfer(client.Asset.GetChain().Chain)
+	for _, node := range affectedNodes {
+
+		if node.ModifiedNode != nil {
+			modifiedNode := node.ModifiedNode
+			if modifiedNode.LedgerEntryType == "AccountRoot" {
+
+				if node.ModifiedNode.FinalFields == nil {
+					continue
+				}
+
+				var (
+					contract     string
+					finalBalance float64
+					parseErr     error
+				)
+
+				if node.ModifiedNode.FinalFields.Balance.XRPAmount != "" {
+					contract = ""
+				} else {
+					finalBalance, parseErr = strconv.ParseFloat(node.ModifiedNode.FinalFields.Balance.TokenAmount.Value, 10)
```

Comment:
please use `xc.AmountHuman` to parse as a decimal.  Using native floating point isn't safe as there can be truncation and rounding errors.

A float64 can safely store 15-17 decimals, which in theory would be enough for XRP.  But on other chains, there are assets that use more decimals/precision.  Let's stay consistent and use a safe decimal type (`xc.AmountHuman` in our case).

https://en.wikipedia.org/wiki/Double-precision_floating-point_format

Also if we do any math with floats, we can get small errors which is bad for and financial reporting.

https://stackoverflow.com/questions/3730019/why-not-use-double-or-float-to-represent-currency